### PR TITLE
Update required docker version to 1.12, including oc cluster up

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

**Comment by jim-minter** [comment]
[test]

**Comment by jim-minter** [comment]
@csrwng @bparees PTAL
@openshift/devex

**Comment by bparees** [comment]
defer to @csrwng

File: pkg/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -251,11 +250,7 @@ func (h *Helper) Start(opt *StartOptions, out io.Writer) (string, error) {
 		env = append(env, "OPENSHIFT_CONTAINERIZED=false")
 	} else {
 		binds = append(binds, "/:/rootfs:ro")
-		propagationMode := ""
-		if opt.SetPropagationMode {
-			propagationMode = ":rslave"
-		}
-		binds = append(binds, fmt.Sprintf("%[1]s:%[1]s%[2]s", opt.HostVolumesDir, propagationMode))
+		binds = append(binds, fmt.Sprintf("%[1]s:%[1]s%[2]s", opt.HostVolumesDir, ":rslave"))
```
Comment (csrwng) [inline pkg/bootstrap/docker/openshift/helper.go]: just a nit, since the propagation mode is now fixed, you can just make it part of the format string.

**Comment by bparees** [comment]
> This means that even though you specify an older version of openshift with the --version parameter (which could presumably run on an older Docker), if you have a newer oc binary, we won't let you.

how do we feel about making it a warning instead of an error?

**Comment by csrwng** [comment]
> how do we feel about making it a warning instead of an error?

I'm fine with that, but a warning is likely to go unnoticed until we fix https://github.com/openshift/origin/issues/12715

**Comment by csrwng** [comment]
@bparees @jim-minter not sure how useful this check is going to be going forward given the new versioning scheme:
https://docs.docker.com/docker-for-mac/release-notes/

**Comment by jim-minter** [comment]
@csrwng bumped - now a warning rather than an error.  I think this check is better than nothing - as things stand if the user is using a CE version number, we don't warn and that's fine (logically it should work as the codebase is newer than 1.12).  If we see they are running < 1.12, we warn.

**Comment by csrwng** [comment]
LGTM [merge]

File: origin.spec [hunk 2]
```diff
@@ -9,7 +9,7 @@
 %global kube_plugin_path /usr/libexec/kubernetes/kubelet-plugins/net/exec/redhat~openshift-ovs-subnet
 
 # docker_version is the version of docker requires by packages
-%global docker_version 1.9.1
+%global docker_version 1.12
```
Comment (stevekuznetsov) [inline origin.spec]: @tdawson is this change already in OCP?

Comment (tdawson) [inline origin.spec]: No, not in 3.5 or 3.6

Comment (stevekuznetsov) [inline origin.spec]: Is OK for this to merge in for 3.5? Do we need to backport it?

**Pull request merged**